src/studio/src/designer/frontend/packages/schema-editor/src/utils.ts

```diff
@@ -2,18 +2,23 @@ import { ILanguage, UiSchemaItem } from './types';
 
 const JsonPointer = require('jsonpointer');
 
-export function getUiSchemaItem(schema: UiSchemaItem[], path: string): UiSchemaItem {
-  let propertyId: string | null = null;
-  if (path.includes('/properties/')) {
-    // eslint-disable-next-line no-param-reassign
-    [path, propertyId] = path.split('/properties/');
-  }
-  let schemaItem: UiSchemaItem = schema.find((item) => item.id === path) || {} as UiSchemaItem;
-  if (schemaItem.properties && propertyId) {
-    schemaItem = schemaItem.properties.find((item: any) => item.id === `${path}/properties/${propertyId}`) || {} as UiSchemaItem;
+function flat(input: UiSchemaItem[] | undefined, depth = 1, stack: UiSchemaItem[] = []) {
+  if (input) {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const item of input) {
+      stack.push(item);
+      if (item.properties instanceof Array && depth > 0) {
+        flat(item.properties, depth - 1, stack);
+      }
+    }
   }
+  return stack;
+}
 
-  return schemaItem;
+export function getUiSchemaItem(schema: UiSchemaItem[], path: string): UiSchemaItem {
+  // should we cache this flattened structure in state somehow for faster lookups?
```

Comment:
Sounds like a good idea!

Reply:
I will create a separate ticket for this optimization